src/renderer_gl.cpp

```diff
@@ -4395,10 +4410,13 @@ namespace bgfx { namespace gl
 			case GL_IMAGE_CUBE:
 			case GL_INT_IMAGE_CUBE:
 			case GL_UNSIGNED_INT_IMAGE_CUBE:
+
+			case GL_SAMPLER_EXTERNAL_OES:
 				if (m_numSamplers < BX_COUNTOF(m_sampler) )
 				{
 					BX_TRACE("Sampler #%d at location %d.", m_numSamplers, loc);
-					m_sampler[m_numSamplers] = loc;
+					GLenum target = gltype == GL_SAMPLER_EXTERNAL_OES ? GL_TEXTURE_EXTERNAL_OES : 0;
```

Comment:
If you're already saving target, if there a good reason why you should not save it for all samplers?

Reply:
Maybe not.  The only reason I did it like that was to only change how the GL_SAMPLER_EXTERNAL_OES was handled and keep the other textures/samplers as before. But the value is only used to bind the texture in TextureGL::commit, so there should not be any problem by saving it for all samplers. I can fix that

Reply:
Or one other reason is that we have to map each sampler to their correct target (e.g. GL_SAMPLER_2D -> GL_TEXTURE_2D). I am not sure how samplers with GL_IMAGE_* should be mapped?

Reply:
I think something like this could work?
```
    GLenum glSamplerTarget(GLenum _sampler){
        switch (_sampler)
        {
        case GL_SAMPLER_2D:
        case GL_INT_SAMPLER_2D:
        case GL_UNSIGNED_INT_SAMPLER_2D:
        case GL_SAMPLER_2D_SHADOW:
            return GL_TEXTURE_2D;
        ...  
                
        case GL_IMAGE_1D:
        case GL_IMAGE_2D:
        ...
            return 0;
        }
        
        BX_CHECK(false, "Unrecognized GL sampler type 0x%04x.", _sampler);
        return 0;
    }
```

Reply:
I pushed a solution where the target is saved for each sampler type, and not just for GL_SAMPLER_EXTERNAL_OES.
I noticed that the AppVeyor build failed, but the errors does not look like they originate from my changes. Or what do you think, @bkaradzic ?